elements-fun/src/confidential.rs

```diff
@@ -23,109 +23,66 @@ use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use std::{fmt, io};
 
 use crate::encode::{self, Decodable, Encodable};
-use crate::issuance::AssetId;
 
 // Helper macro to implement various things for the various confidential
 // commitment types
 macro_rules! impl_confidential_commitment {
-    ($name:ident, $inner:ty, $prefixA:expr, $prefixB:expr) => {
-        impl_confidential_commitment!($name, $inner, $prefixA, $prefixB, |x| x);
-    };
-    ($name:ident, $inner:ty, $prefixA:expr, $prefixB:expr, $explicit_fn:expr) => {
+    ($name:ident, $prefixA:expr, $prefixB:expr) => {
         impl $name {
-            /// Create from commitment.
-            pub fn from_commitment(bytes: &[u8]) -> Result<$name, encode::Error> {
-                if bytes.len() != 33 {
+            pub const fn is_valid_prefix(tag: u8) -> bool {
+                tag == $prefixA || tag == $prefixB
+            }
+
+            pub fn new(tag: u8, commitment: &[u8]) -> Result<Self, encode::Error> {
+                if commitment.len() != 32 {
                     return Err(encode::Error::ParseFailed(
-                        "commitments must be 33 bytes long",
+                        "commitments must be 32 bytes long",
```

Comment:
:question: I know we talked about this yesterday, but the tag is part of the commitment, right? It's a nitpick, but it seems incorrect to say that commitments must be 32 bytes long. I guess the argument `commitment` is actually the commitment's x-coordinate. It may not be worth it to disambiguate though.

Reply:
Fair, it just weird because here I am only deserializing the last 32 bytes :D

I guess this can be an implementation detail though and the error message should be better!